core/include/core/serialization.h

```diff
@@ -72,7 +72,7 @@ _g3_class_version(T *)
 }
 
 #define G3_CHECK_VERSION(v) \
-	if (v > _g3_class_version(this)) \
+	if ((int)v > _g3_class_version(this)) \
```

Comment:
This also looks like a case where the original signed side should be unsigned; [Cereal versions are unsigned](http://uscilab.github.io/cereal/assets/doxygen/group__Utility.html#gabea833652252591a51f75cb9ce31f44a), specifically `uint32_t`, so `_g3_class_version` should probably just be returning that in the first place.